llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCExpr.cpp

```diff
@@ -61,6 +65,21 @@ void AMDGPUVariadicMCExpr::printImpl(raw_ostream &OS,
   case AGVK_Max:
     OS << "max(";
     break;
+  case AGVK_ExtraSGPRs:
+    OS << "extrasgprs(";
+    break;
+  case AGVK_TotalNumVGPRs:
+    OS << "totalnumvgprs(";
+    break;
+  case AGVK_TotalNumVGPRs90A:
+    OS << "totalnumvgprs90a(";
```

Comment:
Shouldn't need a separately named expression for a different behavior on a target?

Reply:
Agree, can't you just parse one string and change the calculation based on the target?